src/externalUser.ts

```diff
@@ -32,18 +37,25 @@ export class ExternalUser {
     /**
      * Creates a new external user
      *
-     * @param _keyManager {KeyManager} - the key manager to use for signing
+     * @param {KeyManager} _keyManager - the key manager to use for signing
+     * @param {StorageFactory} _storageFactory - the storage factory to use for persistent storage
      */
     constructor(_keyManager: KeyManager, _storageFactory: StorageFactory) {
         this.keyManager = _keyManager;
-        this.storage = createStorage<ExternalUserStorage>('tonomy.externalUser.', _storageFactory);
+        this.storage = createStorage<ExternalUserStorage>(STORAGE_NAMESPACE + 'external.user.', _storageFactory);
     }
 
-    static async getUser(
-        keyManager: KeyManager,
-        storageFactory: StorageFactory = browserStorageFactory
-    ): Promise<ExternalUser> {
-        const user = new ExternalUser(keyManager, storageFactory);
+    /**
+     * Retreives the user from persistent storage if it exists and verifies the keys are valid
+     *
+     * @param {StorageFactory} [storageFactory=browserStorageFactory] - the storage factory to use for persistent storage
+     * @param {KeyManager} [keyManager=new JsKeyManager()] - the key manager to use for signing
+     * @returns {Promise<ExternalUser>} - the user
+     */
+    static async getUser(options: { storageFactory: StorageFactory; keyManager: KeyManager }): Promise<ExternalUser> {
+        if (!options.storageFactory) options.storageFactory = browserStorageFactory;
+        if (!options.keyManager) options.keyManager = new JsKeyManager();
```

Comment:
to avoid multiple branching you can use
   options = Object.assign({keymanager: new JsKeyManager(), storageFactory:browserStorageFactory},options)